src/client/components/forms/type-editor/identifier-type.tsx

```diff
@@ -0,0 +1,446 @@
+	const handleModalSubmit = useCallback(() => {
+		if (selectedParentType !== null) {
+			setFormData((prevFormData) => ({
+				...prevFormData,
+				childOrder, parentId: selectedParentType
+			}));
+			setShowModal(false);
+		}
+	}, [formData, childOrder, selectedParentType]);
+
+	const handleInputChange = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+		  const {name, value} = event.target;
+		  setFormData((prevFormData) => ({
+			...prevFormData,
+			[name]: value
+		  }));
+	}, [formData]);
+
+	const handleDeprecatedChange = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+		const {value} = event.target;
+		setFormData((prevFormData) => ({
+			...prevFormData,
+			deprecated: value === 'true'
+		}));
+	}, [formData]);
+
+	const errorAlertClass = classNames('text-center', 'margin-top-1', {'d-none': !showIncompleteFormError});
+
+	const getEntityTypeLabel = useCallback(option => option.name, []);
+
+	const getEntityTypeValue = useCallback(option => option.name, []);
+
+	const getParentTypeValue = useCallback(option => option.id, []);
+
+	// Callback function to format the option label
+	const formatParentTypeOptionLabel = useCallback(option => (
+		<div className="small">
+			<div>
+				<strong>Label:&nbsp;</strong>{option.label}
+			</div>
+			<div>
+				<strong>Description:&nbsp;</strong>{option.description}
+			</div>
+		</div>
+	), []);
+
+	const handleEntityTypeChange = useCallback((selectedOption) => {
+		if (selectedOption) {
+			setFormData({...formData, entityType: selectedOption.name});
+		}
+		else {
+			setFormData({...formData, entityType: null});
+		}
+	}, [formData]);
+
+	function isValid() {
+		return Boolean(formData.entityType);
+	}
+
+	const handleSubmit = useCallback(async (event: FormEvent<HTMLFormElement>) => {
+		event.preventDefault();
+		if (!isValid()) {
+			setShowIncompleteFormError(true);
+			return;
+		}
+		if (!isFormEdited) {
+			// Display an error message if no changes were made
+			setNoChangesError(true);
+			// Hide the error message after 3 seconds
+			setTimeout(() => {
+			  setNoChangesError(false);
+			}, 3000);
+			return;
+		}
+
+		let submissionURL;
+		if (identifierTypeData.id) {
+			submissionURL = `/identifier-type/${identifierTypeData.id}/edit/handler`;
+		}
+		else {
+			submissionURL = '/identifier-type/create/handler';
+		}
+
+		try {
+			await request.post(submissionURL)
+				.send(formData);
+			window.location.href = '/identifier-types';
+		}
+		catch (err) {
+			throw new Error(err);
+		}
+	}, [formData, isFormEdited, showIncompleteFormError]);
+
+	// When we change EntityType, then we must check the validity of the parentType
+	// in case it is already selected
+	useEffect(() => {
+		if (formData.parentId) {
+			const parentType = parentTypes.find(type => type.id === formData.parentId);
+			if (formData.entityType !== parentType.entityType) {
+				handleRemoveParent();
+			}
+		}
+	}, [formData.entityType]);
```

Comment:
I dont think this is required.

Initially, when there is no entityType selected we show the entire list of identifierTypes as possible parentTypes. When we select an entityType, then we filter this list of identifierTypes so that all possible parentTypes have the same entityType. Now, even if we change parentId from here on, the list will remain the same since there is no change in the entityType.